src/jit/codegenxarch.cpp

```diff
@@ -1813,162 +1813,20 @@ CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
                     assert(structDesc.eightByteCount == CLR_SYSTEMV_MAX_EIGHTBYTES_COUNT_TO_PASS_IN_REGISTERS);
 
                     regNumber retReg0 = REG_NA;
-                    emitAttr size0 = EA_UNKNOWN;
-                    unsigned offset0 = structDesc.eightByteOffsets[0];
+                    unsigned __int8 offset0 = 0;
```

Comment:
How about abstracting the logic to return a struct in two return registers in a separate codegen routine genStructReturn()?  GT_RETURN codegen has become quite complex with lot of #ifdefs we should take this opportunity to clean it up.  I would suggest we organize it as follows in a seperate routine genReturn()

#ifdef _TARGET_X86_
if (treeNode->TypeGet() == TYP_LONG)
{
  // X86  specific code for return long in two registers
}
else if (varTypeIsFloating(treeNode)
{
   // x86 float return
}
else
#endif  // end of x86 specific logic
{
   if (IsStructReturn(treeNode))
   {
         // case of either retbuf returned in RAX or a struct returned in two registers
         // IsStructReturn() returns false on amd64/x86 windows always
         genStructReturn(treeNode);
   }
   else
   {
        // Logic common to x86/amd64 windows/amd64 unix
   }   
}